lib/persist-state.c

```diff
@@ -267,6 +314,12 @@ _alloc_value(PersistState *self, guint32 orig_size, gboolean in_use, guint8 vers
   persist_state_unmap_entry(self, self->current_ofs);
 
   self->current_ofs += size + sizeof(PersistValueHeader);
+
+  if (!_check_watermark(self) && !_grow_store(self, self->current_size + PERSIST_FILE_INITIAL_SIZE))
+    {
+      main_loop_exit();
+    }
+
   return result;
```

Comment:
still, it is still pulling in the entire syslog-ng core lib as a dependency.

-- 
Bazsi

On Mon, Nov 30, 2015 at 12:58 PM, juhaszviktor <notifications@github.com>
wrote:

> In lib/persist-state.c
> <https://github.com/balabit/syslog-ng/pull/717#discussion_r46134848>:
>
> > @@ -267,6 +314,12 @@ _alloc_value(PersistState *self, guint32 orig_size, gboolean in_use, guint8 vers
> >    persist_state_unmap_entry(self, self->current_ofs);
> >
> >    self->current_ofs += size + sizeof(PersistValueHeader);
> > +
> > +  if (!_check_watermark(self) && !_grow_store(self, self->current_size + PERSIST_FILE_INITIAL_SIZE))
> > +    {
> > +      main_loop_exit();
> > +    }
> > +
> >    return result;
>
> My idea is, that I will extend the apphook with an
> apphook_request_shutdown and its state AH_REQUEST_SHUTDOWN. The main loop
> can subscribe for this.
> @bazsi <https://github.com/bazsi>, what is your opinion about this idea?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balabit/syslog-ng/pull/717/files#r46134848>.
>